### PR TITLE
Expand the guide for contributors

### PR DESCRIPTION
**Pull request by mik-laj**
Hello,

Previous documentation was problematic. I expanded it to make it more clear.

I have also fixed one compatibility problem with Debian. Everything was tested before on MacOS, and now I've checked everything on Debian.

Best regards
Kamil

### CODE REVIEW

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -54,49 +55,95 @@
     └── sphinx_airflow_theme
 ```
 
-## Working with the project
+# Working with the project
+
+Work with the site and documentation requires that your computer be properly prepared. Most tasks can
+be done by the site.sh script
+
+### Prerequisite Tasks
```
Comment (louisguitton) [inline CONTRIBUTE.md]: Mentioning pre-commit would be helpful for people not used to it.
```sh
brew install pre-commit
pre-commit install
```

Comment (mik-laj) [reply CONTRIBUTE.md]: Thanks. I added section.

File: CONTRIBUTE.md [hunk 2]
```diff
@@ -17,9 +17,10 @@
  under the License.
 -->
 
-## General directory structure
+Contrubitor Guide
```
Comment (Fokko) [inline CONTRIBUTE.md]:
```suggestion
Contributor Guide
```

File: CONTRIBUTE.md [hunk 3]
```diff
@@ -54,49 +55,120 @@
...
+```bash
+pre-commit install
+```
+
+To run all checks on your staged files, enter:
+```bash
+pre-commit run
+```
+
+To run all checks on all files by using, enter:
```
Comment (mik-laj) [inline CONTRIBUTE.md]:
```suggestion
To run all checks on all files, enter:
```

**Comment by mik-laj** [comment]
Thanks @kaxil

**Pull request merged**